File: tasks/implementations/node/tasks/matrix_multiplication.js
const fs = require('fs');
module.exports = function(size, fixturesRoot) {
  const tokens = fs.readFileSync(`${fixturesRoot}/generated/matrix/${size}.txt`, 'utf8').trim().split(/\s+/).map(Number);
  const n = tokens[0];
  const nums = tokens.slice(1);
  const split = n * n;
  const a = nums.slice(0, split);
  const b = nums.slice(split);
  let total = 0;
  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      let cell = 0;
      for (let k = 0; k < n; k++) {
        cell += a[i * n + k] * b[k * n + j];
      }
      total += cell;
    }
  }
  console.log(total);
};

File: tasks/implementations/rust/tasks/matrix_multiplication.rs
use std::fs;

pub fn run(size: &str, fixtures_root: &str) {
    let text = fs::read_to_string(format!("{}/generated/matrix/{}.txt", fixtures_root, size)).unwrap();
    let tokens: Vec<i64> = text.split_whitespace().map(|x| x.parse::<i64>().unwrap()).collect();
    let n = tokens[0] as usize;
    let nums = &tokens[1..];
    let split = n * n;
    let a = &nums[..split];
    let b = &nums[split..];
    let mut total = 0i64;
    for i in 0..n {
        for j in 0..n {
            let mut cell = 0i64;
            for k in 0..n {
                cell += a[i * n + k] * b[k * n + j];
            }
            total += cell;
        }
    }
    println!("{}", total);
}
